src/libraries/System.Net.Mail/src/System/Net/Mime/HeaderCollection.cs

```diff
@@ -36,17 +33,6 @@ public override void Remove(string name)
                 throw new ArgumentException(SR.Format(SR.net_emptystringcall, nameof(name)), nameof(name));
             }
 
-            MailHeaderID id = MailHeaderInfo.GetID(name);
```

Comment:
I don't undersatnd the changes in this file.

Reply:
My assumptions:

* `_part` is always null, so the if and else if both always fail in the `&&`, so the bodies never execute.
* The left arguments to the `&&` don't look side-effecty, so remove them.
* `id` is unreferenced.  As long as MailHeaderInfo.GetID is side-effect-free, remove the call.

Reply:
fair enough

Reply:
I guess since he deleted the variable outright, it's not an "assumption" that it was always null... as long as everything compiles :smile:.

Reply:
It's what Jeremy said. _part was being explicitly initialized to null; when I removed that, it unsuppressed the warning about the field never being set, because it wasn't being set anywhere else, highlighting that it would always be null. Thus all comparisons against null could be treated as constant true or false expressions, and the resulting dead code removed. GetID here is just a TryGetValue on a dictionary, and is non-side-effecting. This all highlights I think arguably the biggest benefit of the new rule: it helps uncover dead code.